### PR TITLE
Pfx

### PR DESCRIPTION
**Pull request by degenerated1123**
Most of the Particle-FX core is implemented. A few types and properties are still not implemented or ignored, but basic functionality is there.

### CODE REVIEW

File: src/logic/visuals/PfxVisual.cpp [hunk 1]
```diff
@@ -126,95 +128,32 @@ void Logic::PfxVisual::onUpdate(float deltaTime)
     }
 
     // Update particle values
-    for(Particle& p : m_Particles)
+    for(Components::PfxComponent::Particle& p : pfx.m_Particles)
         updateParticle(p, deltaTime);
 
-    for(int i=0;i<static_cast<int>(m_Particles.size());i++)
+    for(int i=0;i<static_cast<int>(pfx.m_Particles.size());i++)
```
Comment (Aquaz) [inline src/logic/visuals/PfxVisual.cpp]: We could use size_t instead of int there and remove cast.

File: src/render/WorldRender.cpp [hunk 2]
```diff
@@ -472,3 +451,84 @@ void Render::debugDrawPath(const World::Waynet::WaynetInstance& waynet, const st
...
+	// Cache
+	static std::vector<Meshes::WorldStaticMeshVertex> quadVertices;
+
+
+	Math::float3 right = config.state.cameraWorld.Rotate(Math::float3(1,0,0)).normalize() * -0.5f; // 0.5 because they get extended into both directions. We want size 1 in total.
+	Math::float3 up = config.state.cameraWorld.Rotate(Math::float3(0,1,0)).normalize() * 0.5f;
+
+	quadVertices.resize(pfx.m_Particles.size() * 6);
+
+	Meshes::WorldStaticMeshVertex test[6];
```
Comment (Aquaz) [inline src/render/WorldRender.cpp]: This can be removed, right?

File: src/logic/PfxManager.cpp [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    if(!loadParticleFXDAT())
+        return false;
+
+    return m_pVM->getDATFile().hasSymbolName(name);
+}
+
+const Logic::PfxManager::Emitter& Logic::PfxManager::getParticleFX(const std::string& name)
+{
+    if(!loadParticleFXDAT())
+        return Logic::PfxManager::Emitter();
```
Comment (degenerated1123) [inline src/logic/PfxManager.cpp]: Right, forgot about that.

File: src/render/WorldRender.cpp (hunk 2 above)
Comment (degenerated1123) [inline src/render/WorldRender.cpp]: The test vertices? Yes they can

Comment (Aquaz) [inline src/render/WorldRender.cpp]: Yeah I meant test arr, just spotted it wasn't used anywhere other than in the memset line below

**Comment by degenerated1123** [comment]
@Aquaz How about now?

**Pull request merged**